Assets/Scripts/navigation/Navigation.cs

```diff
@@ -0,0 +1,186 @@
+		//private LineRenderer lineRenderer;
+		public TMP_Text text;
+		public Image ImageSlot;
+		private const string ResourcePath = "crew_portraits";
+		private const string DefaultPortrait = "crew_portraits/phoenician_sailor";
+		public GameObject menu;
+		public GameObject Navgater;
+		private bool StartNavigation = false;
+		Vector3 nexPoint;
+		private void Start() {
+			cities = new city();
+			menu.SetActive(false);
+			Navgater.SetActive(false);
+			if (AI == null) {
+				AI = GetComponent<NavMeshAgent>();
+			}
+			//if(lineRenderer == null) {
+			//	lineRenderer = GetComponent<LineRenderer>();
+			//}
+		}
+		private void Update() {
+
+			if(!TitleScreen.activeSelf && StartNavigation) {
+				ShowMenu(StartNavigation);
+			}
+		}
+		public static float CalcAngle(Vector3 from, Vector3 to) {
+			Vector3 delta = from - to;
+			return Mathf.Atan2(delta.x, delta.z) * 180 / Mathf.PI;
+		}
+		public static float EulerNormalize(float angle) {
+			while (angle < -180) angle += 360;
+			while (angle > 180) angle -= 360;
+			return angle;
+		}
+		public void Setdestination(string target, int ID) {
+			if (target == null) {
+				Debug.LogError("Didn't find the city");
+				return;
+			}
+			Vector3 targetlocaion = cities.GetCityLocation(target);
+			if (targetlocaion == null) {
+				Debug.LogError("Unable to find city location");
+				return;
+			}
+			//assign image
+			ImageSlot.sprite = Resources.Load<Sprite>(ResourcePath + "/" + ID) ?? Resources.Load<Sprite>(DefaultPortrait);
+			StartNavigation = true;
+			//find player
+			player = GameObject.Find("playerShip").transform.Find("ShipParent").transform.Find("kyrenia_parent").transform;
+			this.transform.position = player.position;
+			float distance = Vector3.Distance(targetlocaion, player.position);
+			float radius = 2;
+			if (distance >= radius) {
+				string Saildirecation = FindPlayerDirection(player); // find what angle is player faceing
+				string Citydirecation = FindCityDirection(targetlocaion,player); // find city angle by the world
+				string WhereToSail = getAIPath(targetlocaion, radius); // find the path
+				text.text = Citydirecation +Saildirecation + WhereToSail; // assign text
+			}
+			if (distance < radius) {
+				text.text = "You have arrive to " + target + ".";
+			}
+		}
+		string FindCityDirection(Vector3 targetlocaion, Transform current) {
+			float angle = CalcAngle(current.position + -current.up, targetlocaion);    // target angle relative to world
+			angle -= 180;
+			string citydirecation = "The city is in the "+ FindDirection(angle)+".";
+			return citydirecation;
+		}
+		string FindPlayerDirection(Transform current) {
+			float angle2 = CalcAngle(current.position, current.position + current.up); // player angle relative to world
+			string Saildirecation = "The Ship is sail to the " + FindDirection(angle2) + ".";
+			return Saildirecation;
+		}
+		string getAIPath(Vector3 target,float radius) {
+			AI.stoppingDistance = radius;
+			AI.SetDestination(target);
```

Comment:
NavMeshAgent.SetDestination is asynchronous, so "the path may not become available until after a few frames later" according to the docs:
https://docs.unity3d.com/ScriptReference/AI.NavMeshAgent.SetDestination.html

i think the code is working because your return "" line will make it ignore the path until the first path is ready (path length <= 1), and then you'll be using the previous path until the next one is computed each time, but maybe you should leave a comment saying that it's doing that, or better use the the NavMeshAgent.pathPending boolean to make the check more explicit